manim/mobject/opengl/opengl_mobject.py

```diff
@@ -2715,45 +2836,50 @@ def check_data_alignment(self, array, data_key):
             )
         return self
 
-    def get_resized_shader_data_array(self, length):
+    def get_resized_shader_data_array(self, length: float) -> np.ndarray:
         # If possible, try to populate an existing array, rather
         # than recreating it each frame
         points = self.points
         shader_data = np.zeros(len(points), dtype=self.shader_dtype)
         return shader_data
 
-    def read_data_to_shader(self, shader_data, shader_data_key, data_key):
+    def read_data_to_shader(
+        self,
+        shader_data: np.ndarray,  # has structured data type, ex. ("point", np.float32, (3,))
+        shader_data_key: str,
+        data_key: str,
+    ) -> None:
         if data_key in self.locked_data_keys:
             return
         self.check_data_alignment(shader_data, data_key)
         shader_data[shader_data_key] = self.data[data_key]
 
-    def get_shader_data(self):
+    def get_shader_data(self) -> np.ndarray:
         shader_data = self.get_resized_shader_data_array(self.get_num_points())
         self.read_data_to_shader(shader_data, "point", "points")
         return shader_data
 
-    def refresh_shader_data(self):
+    def refresh_shader_data(self) -> None:
         self.get_shader_data()
 
-    def get_shader_uniforms(self):
+    def get_shader_uniforms(self) -> dict[str, Any]:
         return self.uniforms
 
-    def get_shader_vert_indices(self):
+    def get_shader_vert_indices(self) -> list[int]:
         return self.shader_indices
 
     @property
-    def submobjects(self):
+    def submobjects(self) -> list[OpenGLMobject]:
```

Comment:
```suggestion
    def submobjects(self) -> Sequence[OpenGLMobject]:
```
Because `list` is invariant, this would cause mypy errors in a bunch of places (in the future, once modules stop being ignored) about it being incompatible.

Reply:
Huh, didn't know about that one. I made the respective changes.